packages/svelte/src/internal/client/dom/elements/attributes.js

```diff
@@ -515,11 +515,54 @@ export function handle_lazy_img(element) {
 	// templates.
 	if (!hydrating && element.loading === 'lazy') {
 		var src = element.src;
+		var loading = element.loading;
 		// @ts-expect-error
 		element[LOADING_ATTR_SYMBOL] = null;
 		element.loading = 'eager';
 		element.removeAttribute('src');
+
+		// since we are removing src before the first animation frame if someone
+		// access src or loading on mount or in an action they will have the wrong value
+		// so we define those two properties so that if accessed through JS it will
+		// return the correct value. Since we can't undo this we also mimic the behavior
+		// of setting the attribute in the setter
+		define_property(element, 'src', {
+			get() {
+				return src;
```

Comment:
This isn't correct? It doesn't take into account relative pathing AFAIK which differs from the attribute value

Reply:
```js
console.log(img.getAttribute("src")); // "images/foo.png"
console.log(img.src);                 // "http://example.com/path/images/foo.png"
```

Reply:
Ugh...I knew something was gonna bit me 😁 I'll check it I can find a fix